### ISSUE TITLE
Streams API no longer accepts resolution="all"

### ISSUE THREAD

**Issue by djcunningham0** [stravalib/stravalib#233]
"all" is not recognized as a valid value for the "resolution" parameter in the Streams API. When I make a request with resolution=all I get a 400 error. It still works for values of "high", "medium", and "low" (high seems to be the default now).

For what it's worth, the "resolution" parameter has been removed from the [official docs](https://developers.strava.com/docs/reference/#api-Streams-getActivityStreams), so maybe it will be completely removed in the future.

The loss of this parameter means it's impossible to get full resolution streams for long activities. Does anyone know if there's another way of getting that data?

https://github.com/hozn/stravalib/blob/b8d1181e32248c6570f0adfa995cb6c7adfccf62/stravalib/client.py#L1190-L1191

**Comment by lwasser** [comment]
hey there @djcunningham0  :wave: i'm going through some of our older issues and cleaning things up. 

[here is where the new streams method is defined](https://github.com/stravalib/stravalib/blob/main/src/stravalib/client.py#L1519) in our current API. 

and [in the strava docs it does show that a resolution is returned](https://developers.strava.com/docs/reference/#api-Streams-getActivityStreams)
<img width="1324" alt="Screenshot 2024-02-11 at 7 08 41 PM" src="https://github.com/stravalib/stravalib/assets/7649194/8305a9b8-f18e-42c4-adae-5b9431c462aa">

which is also represented in [our strava_model.py ](https://github.com/stravalib/stravalib/blob/main/src/stravalib/strava_model.py#L93)
 module which is generated from the strava swagger.json file. As such it looks to me as if there is no resolution = all option available. our model allows for low, medium and high as you mention above.  

does the "high" option not work for you? What gets returned when you try that? a filtered version of the activity? 

Please let us know if this is still an issue. 

if i don't hear from you in a week i will close this.

but if it's an ongoing issue i'm happy to test it out one some longer activities just to see what happens. I know i have been able to download the streams for long activities but i never looked closely at the point density / resolution.

**Comment by dannycunningham-8451** [comment]
It looks like this is no longer an issue. The solution to my original issue is to not any value for the "resolution" parameter. If I don't specify resolution, it returns every single data point in the stream. (For comparison, if I specify resolution="high", then it trims condenses long activities down to ~8500 data points.) I don't know if the functionality changed since I opened the issue or if I didn't notice the solution at the time, but that doesn't really matter going forward.

Some functionality and documentation could be improved (described below), but I'd consider it low priority. This issue could be closed since there's a way to get full resolution detail and it only affects very long activities anyway.

---

Inconsistencies:

(I'm testing with activity 7595268894, for reference.)

When I hit the API endpoint at /activities/{activity_id}/streams and specify the parameter **resolution="high"**, the response object has the following attributes:
* resolution = "high"
* original_size = 49,320
* data: <list with 8,892 elements> **-- it did not pull all of the original data points!**

When instead I hit the same endpoint (/activities/{activity_id}/streams) but I **don't specify the resolution parameter**, the response object has the following attributes:
* resolution = "high" -- the same value as the other response, **even though they actually have different resolution!**
* original_size = 49,320
* data: <list with 49,320 elements> -- it contains all of the original data points


Suggestions:
* When the user doesn't supply the resolution parameter in the API request, the response object should have resolution="None" or resolution="all", or some other value that differentiates it from resolution="high".
* The documentation should be updated accordingly.

**Comment by djcunningham0** [comment]
It looks like this is no longer an issue. The solution is to not a value for the "resolution" parameter. If I don't specify resolution, it returns every single data point in the stream. (For comparison, if I specify resolution="high", then it trims condenses long activities down to ~8500 data points.) I don't know if the functionality changed since I opened the original issue or if I didn't notice the solution at the time, but that doesn't really matter going forward.

Some functionality and documentation could be improved (described below), but I'd consider it low priority. This issue could be closed since there's a way to get full resolution detail and it only affects very long activities anyway.

---

Inconsistencies:

(I'm testing with activity 7595268894, for reference.)

When I hit the API endpoint at /activities/{activity_id}/streams and specify the parameter **resolution="high"**, the response object has the following attributes:
* resolution = "high"
* original_size = 49,320
* data: <list with 8,892 elements> **-- it did not pull all of the original data points!**

When instead I hit the same endpoint (/activities/{activity_id}/streams) but I **don't specify the resolution parameter**, the response object has the following attributes:
* resolution = "high" -- the same value as the other response, **even though they actually have different resolution!**
* original_size = 49,320
* data: <list with 49,320 elements> -- it contains all of the original data points


Suggestions:
* When the user doesn't supply the resolution parameter in the API request, the response object should have resolution="None" or resolution="all", or some other value that differentiates it from resolution="high".
* The documentation should be updated accordingly.

**Comment by lwasser** [comment]
hey @djcunningham0  👋  Thanks for coming back to this (old!) issue. This is unexpected behavior. Perhaps the way strava treats resolution is that you either want filtered data (filtered at whatever low, medium high thresholds they use) or not filtered (ie None / no filtering). 

i also see in our code that this parameter is [actually deprecated](https://github.com/stravalib/stravalib/blob/main/src/stravalib/client.py#L1533) with a note about it being undocumented and thus actually unstable. 

Would a statement in the docstring about how this works (noting that it is technically unsupported by Strava) above be helpful ? Our code is typed in a way that suggests the values low, medium, high and None are acceptable for this request. 

`resolution: Literal["low", "medium", "high"] | None = None,`

But it isn't clear what  `resolution=None` (or just no value provided for that param) returns as of now. 

Would you be interested in submitting a small docstring clarification? 

Maybe something like:

```python
        resolution : str, optional
            Indicates desired number of data points. 'low' (100), 'medium'
            (1000) or 'high' (10000). A value of None will return full resolution data.
            .. deprecated::

                This param is not officially supported by the Strava API and
                may be removed in the future.
```

**Comment by lwasser** [comment]
@all-contributors please add @djcunningham0 for docs, ideas

**Comment by lwasser** [comment]
hey @djcunningham0 i had some more time to play with this - i've updated the docs here:

https://github.com/stravalib/stravalib/pull/581 if you want to have a look at this pr. I am not sure however yet how resolution works. I noticed for a super short activity where i walked like a mile or two - medium and high resolution are the same. I didn't see a different between high and resolution - None ("all") even for a slightly longer hike - 5ish miles.

now it was slow moving. i could look at another longer activity like a super long run to see if there is a difference between high and resolution - None.  But so far i don't see a difference.   Please have a look at that PR and let me know what you think. i'll close this issue once the pr is merged.

**Comment by djcunningham0** [comment]
When I test with a very long activity (for example, [12082203751](https://www.strava.com/activities/12082203751) with elapsed time over 10 hours), I get a shorter list of data points when I specify "high" resolution compared to when I don't specify resolution at all. 

Here are the requests I'm making to the Strava API (`client` is an OAuth client authenticated with my credentials):

<img width="624" alt="image" src="https://github.com/user-attachments/assets/f378361c-d59e-4f02-9eb1-1f68d62d107d">

**Comment by lwasser** [comment]
ok @djcunningham0 IF you have a moment, have a look at #588 and let me know if this clarifies things a bit (or suggest edits).

**Comment by djcunningham0** [comment]
@lwasser looks good!
